src/RuleCollection.php

```diff
@@ -669,9 +669,10 @@ public function showListRules($target, $options = [])
             $url = $CFG_GLPI["root_doc"];
         }
 
-       //if rules provides an initRules method, then we're able to reset them
-        if (method_exists($this->getRuleClass(), 'initRules')) {
-            echo "<a class='btn btn-primary' id='reset_rules' href='" . $rule->getSearchURL() . "?reinit=true&subtype=" . $this->getRuleClassName() . "' " .
+        // if rules provides has default rules, then we're able to reset them
+        $ruleclass = $this->getRuleClass();
+        if ($ruleclass instanceof Rule && $ruleclass->hasDefaultRules()) {
```

Comment:
instanceof test required ? I though getRuleClass method will always return a Rule class, no ?

Reply:
Result could be `null` too.